File: Python/1828.py
# 1828. Queries on Number of Points Inside a Circle
# https://leetcode.com/problems/queries-on-number-of-points-inside-a-circle/

class Solution:
  def countPoints(self, points: List[List[int]], queries: List[List[int]]) -> List[int]:
    out = []

    for _, q in enumerate(queries):
      cnt = 0

      for _, p in enumerate(points):
        if ((pow(p[0] - q[0], 2) + pow(p[1] - q[1], 2)) <= pow(q[2], 2)):
          cnt += 1

      out.append(cnt)

    return out

File: Rust/1828.rs
/**
 * 1828. Queries on Number of Points Inside a Circle
 * https://leetcode.com/problems/queries-on-number-of-points-inside-a-circle/
 */

impl Solution {
  pub fn count_points(points: Vec<Vec<i32>>, queries: Vec<Vec<i32>>) -> Vec<i32> {
    let mut out: Vec<i32> = Vec::new();

    for q in queries.iter() {
      let mut cnt: i32 = 0;

      for p in points.iter() {
        if (p[0] - q[0]).pow(2) + (p[1] - q[1]).pow(2) <= q[2].pow(2) {
          cnt += 1;
        }
      }

      out.push(cnt);
    }

    return out;
  }
}
